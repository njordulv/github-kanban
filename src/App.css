* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

html,
body {
  background-color: #22272e;
  font-family: 'Open Sans', sans-serif;
}

ul {
  list-style: none;
}

.columnAddItem {
  position: relative;
}

.columnAddItem:before {
  position: absolute;
  content: '';
  top: 11px;
  left: 11px;
  right: 11px;
  height: 93px;
  border-radius: 5px;
  z-index: 0;
  border: 1px inset rgb(68, 76, 86);
  opacity: 0.6;
}

.columnAddItem:after {
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  content: '\2295';
  font-size: 54px;
  top: 30px;
  left: 50%;
  margin-left: -27px;
  height: 54px;
  width: 54px;
  opacity: 0.1;
}
