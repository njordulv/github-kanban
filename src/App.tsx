import { ChakraProvider, Box, Text, Grid, GridItem } from '@chakra-ui/react'
import Form from './components/Form'
import styles from './styles/app.module.scss'

export const App = () => (
  <ChakraProvider>
    <Box textAlign="center" fontSize="xl" className={styles.app}>
      <GridItem p="7" bg="#1d2128" className={styles.header}>
        <Form />
      </GridItem>
      <Grid className={styles.content}>
        <GridItem rounded={5}>
          <Grid templateColumns="repeat(3, 1fr)" className={styles.content__boxes}>
            <GridItem className={styles.content__box}>
              <Text className={styles.content__heading}>To Do</Text>
              <Box className={styles.content__item}>
                <Text>items</Text>
              </Box>
            </GridItem>
            <GridItem className={styles.content__box}>
              <Text className={styles.content__heading}>In Progress</Text>
              <Box className={styles.content__item}>
                <Text>items</Text>
              </Box>
            </GridItem>
            <GridItem className={styles.content__box}>
              <Text className={styles.content__heading}>Done</Text>
              <Box className={styles.content__item}>
                <Text>items</Text>
              </Box>
            </GridItem>
          </Grid>
        </GridItem>
      </Grid>
    </Box>
  </ChakraProvider>
)
