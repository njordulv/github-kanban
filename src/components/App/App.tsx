import { Box, Grid, GridItem } from '@chakra-ui/react'
import AppProviders from 'providers/AppProviders'
import Header from 'components/Header'
import Board from 'components/Board'
import 'components/App/App.css'

export const App = () => (
  <AppProviders>
    <Box bg="#22272e">
      <Grid
        textAlign="center"
        fontSize="xl"
        color="#c5d1de"
        m="auto"
        w="100%"
        p={0}
        maxW="1170px"
        minH="calc(100vh - 275px)"
      >
        <GridItem px={7} py={14} position="relative">
          <Header />
        </GridItem>
        <GridItem px={7} rounded={5}>
          <Box color="#c5d1de">
            <Board />
          </Box>
        </GridItem>
      </Grid>
    </Box>
  </AppProviders>
)
