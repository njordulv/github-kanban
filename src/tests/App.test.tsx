import { render, screen } from '@testing-library/react'
import { App } from 'components/App/App'

describe('App component', () => {
  it('Renders "Done" & "Progress" texts', () => {
    render(<App />)

    const DoneElement = screen.getByText(/Done/i)
    const ProgressElement = screen.getByText(/Progress/i)

    expect(DoneElement).toBeInTheDocument()
    expect(ProgressElement).toBeInTheDocument()
  })

  it('If renders correctly', () => {
    const AppSnapshot = render(<App />)

    expect(AppSnapshot).toMatchSnapshot()
  })
})
