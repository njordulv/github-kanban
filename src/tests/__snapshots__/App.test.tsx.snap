// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component If renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-0"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div>
      <div
        class="css-wa68tp"
      >
        <div
          class="css-1dcuefj"
        >
          <div
            class="css-1yi0olm"
          >
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <form
                class="css-1m42ln8"
              >
                <input
                  class="chakra-input css-102u4x6"
                  id="field-:r1:"
                  placeholder="Enter repo URL"
                  type="text"
                  value=""
                />
                <button
                  class="chakra-button css-19tgw0z"
                  type="submit"
                >
                  Load issues
                </button>
              </form>
            </div>
          </div>
          <div
            class="css-pbew5c"
          >
            <div
              class="css-5dlt6v"
            >
              <div
                class="css-xnby5e"
              >
                <div
                  class="css-116cwdx"
                >
                  <h3
                    class="chakra-heading css-5ms7ou"
                  >
                    To Do
                  </h3>
                  <div
                    class="columnAddItem css-hrbizy"
                    data-rbd-droppable-context-id="1"
                    data-rbd-droppable-id="toDo"
                  />
                </div>
                <div
                  class="css-116cwdx"
                >
                  <h3
                    class="chakra-heading css-5ms7ou"
                  >
                    In Progress
                  </h3>
                  <div
                    class="columnAddItem css-hrbizy"
                    data-rbd-droppable-context-id="1"
                    data-rbd-droppable-id="inProgress"
                  />
                </div>
                <div
                  class="css-116cwdx"
                >
                  <h3
                    class="chakra-heading css-5ms7ou"
                  >
                    Done
                  </h3>
                  <div
                    class="columnAddItem css-hrbizy"
                    data-rbd-droppable-context-id="1"
                    data-rbd-droppable-id="done"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-label="Notifications-top"
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-top-left"
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-top-right"
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom-left"
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom"
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-bottom-right"
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-1"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      id="rbd-hidden-text-1-hidden-text-4"
      style="display: none;"
    >
      
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

    </div>
  </body>,
  "container": <div>
    <div
      class="css-wa68tp"
    >
      <div
        class="css-1dcuefj"
      >
        <div
          class="css-1yi0olm"
        >
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <form
              class="css-1m42ln8"
            >
              <input
                class="chakra-input css-102u4x6"
                id="field-:r1:"
                placeholder="Enter repo URL"
                type="text"
                value=""
              />
              <button
                class="chakra-button css-19tgw0z"
                type="submit"
              >
                Load issues
              </button>
            </form>
          </div>
        </div>
        <div
          class="css-pbew5c"
        >
          <div
            class="css-5dlt6v"
          >
            <div
              class="css-xnby5e"
            >
              <div
                class="css-116cwdx"
              >
                <h3
                  class="chakra-heading css-5ms7ou"
                >
                  To Do
                </h3>
                <div
                  class="columnAddItem css-hrbizy"
                  data-rbd-droppable-context-id="1"
                  data-rbd-droppable-id="toDo"
                />
              </div>
              <div
                class="css-116cwdx"
              >
                <h3
                  class="chakra-heading css-5ms7ou"
                >
                  In Progress
                </h3>
                <div
                  class="columnAddItem css-hrbizy"
                  data-rbd-droppable-context-id="1"
                  data-rbd-droppable-id="inProgress"
                />
              </div>
              <div
                class="css-116cwdx"
              >
                <h3
                  class="chakra-heading css-5ms7ou"
                >
                  Done
                </h3>
                <div
                  class="columnAddItem css-hrbizy"
                  data-rbd-droppable-context-id="1"
                  data-rbd-droppable-id="done"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
